packages/gatsby-theme-newrelic/src/components/MarkdownContainer.js

```diff
@@ -39,13 +39,13 @@ const MarkdownContainer = ({
           margin-bottom: var(--text-spacing);
         }
 
-        h2:not(:first-child) {
+        h2:not(:first-of-type) {
```

Comment:
I know the warnings are annoying, but in general, be careful with these changes. We want to make sure to test them thoroughly, otherwise you can end up with weird bugs. `first-child` and `first-of-type` are not the same thing. Take for example the following:

```js
<section className="container">
  <div className="divider" />
  <h2>First title</h2>
  <p>Stuff</p>
  <h2>Second title</div>
</div>
```

And the following CSS:

```js
h2 {
  margin-top: 1rem;
}

.container h2:first-child {
  margin-top: 0;
}
```

Given this CSS, we'd expect that the "First title" `h2` would have `1rem` of margin above it, because we've said that we _only_ want to remove it if its the first child in the container.

Now lets change the CSS to this:

```css
h2 {
  margin-top: 1rem;
}

.container h2:first-of-type {
  margin-top: 0;
}
```

Rather than removing the space only when the `h2` is the first child, you are removing the spacing from the first _instance_ of it. If the intention is to keep the space above the `h2` when its not the first child, then this updated CSS has broken that.

I would definitely vet these changes pretty thoroughly as it could collapse/add space where you don't expect.

Reply:
Ah, thanks for catching that. I thought I tested it thoroughly, but it would appear like I've missed some cases.

I might just back this out for now since the real reason I made this PR was to fix the errors, not the warnings.